examples/00-mapdl-examples/spotweld.py

```diff
@@ -0,0 +1,75 @@
+"""
+.. _ref_spotweld:
+
+=================================================
+Running an input file - spotweld SHELL181 example
+=================================================
+
+This MAPDL example demonstrates how to model spot welding on three
+thin sheets of metal. Here, the full input file is simply run using
+the PyMAPDL interface.
+
+Using the following commands, you can directly use an APDL
+script within a PyMAPDL session with the following
+commands:
+
+"""
+
+from ansys.mapdl.core import launch_mapdl
+
+mapdl = launch_mapdl()
+if mapdl._local:
+    mapdl.upload("./spotweld/spot_weld.inp")
+    mapdl.input("spot_weld.inp")
+else:
+    mapdl.input("./spotweld/spot_weld.inp")
```

Comment:
Also, I believe `mapdl.input` is clever enough to upload the file if it does not exists in the remote instance.

See https://github.com/ansys/pymapdl/blob/fdf3a038bfdce4c7434625ea837db8cece8c7700/src/ansys/mapdl/core/mapdl_grpc.py#L1753

Check this, it works, remove the `if`. If it doesn't, we will have to fix it hence open an issue.

Reply:
It was not running on my local instance, this is why I added this `if` condition.

Reply:
I opened this [issue](https://github.com/ansys/pymapdl/issues/2302) to have a closer look to this problem.